terraform/modules/cloud/aws/deploy/fleet/templates/user_data.bash

```diff
@@ -4,11 +4,17 @@
 # set -x
 exec > >(tee /tmp/user-data.log|logger -t user-data ) 2>&1
 
-export INFRASTRUCTURE_ANSIBLE_VAULT_PASSWORD=`aws --region ${region} secretsmanager get-secret-value --secret-id ansible_vault_password --output text --query 'SecretString'`
-
 git clone -b ${bootstrap_version} --single-branch https://github.com/aeternity/infrastructure.git /infrastructure
 cd /infrastructure/ansible
 
+# Temporary fix/workaround for non-executable vault install
+chmod +x /usr/bin/vault
+
+# Authenticate the instance to CSM
+PKCS7=$(curl -s http://169.254.169.254/latest/dynamic/instance-identity/pkcs7 | tr -d '\n')
+export VAULT_ADDR=${vault_addr}
+export VAULT_TOKEN=$(vault write -field=token auth/aws/login pkcs7=$PKCS7 role=${vault_role})
```

Comment:
Do I understand correctly that PKCS7 is associated with identity of AWS organization that instance has been spawned by, and that that is the trust link with vault? (Otherwise anyone could spawn instance and associated it to vault and be able to run monitoring/deploy.yml .)

Reply:
Yes, it's bound to an AWS account and regions. It can be further limited, however considering our current AWS account users, it should be enough.